.github/workflows/linting.yml

```diff
@@ -1,5 +1,7 @@
 name: Linting
 
+permissions: read-all
```

Comment:
There are a lot of fine-grained permissions here, which if set would technically be more secure:
https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

Reply:
Yes, indeed.  I suppose you are referring to the ability to set some of the finer grained permissions to `none` as I think effectively what you have done here is restricted them all to `read`, which is still quite a bit better than the default: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

Reply:
Right. Copy-pasted from the link I posted

> You can define the access that the GITHUB_TOKEN will permit by specifying read, write, or none as the value of the available scopes within the permissions key.

```
permissions:
  actions: read|write|none
  checks: read|write|none
  contents: read|write|none
  deployments: read|write|none
  id-token: read|write|none
  issues: read|write|none
  discussions: read|write|none
  packages: read|write|none
  pages: read|write|none
  pull-requests: read|write|none
  repository-projects: read|write|none
  security-events: read|write|none
  statuses: read|write|none
```
> If you specify the access for any of these scopes, all of those that are not specified are set to none.

Reply:
So, for example, we should be able to do:
```
permissions:
  actions: read
```
And everything else will be `none`.

But that requires knowing exactly which of those permissions the actions need

Reply:
Exactly!

Reply:
Or maybe we should be setting EACH read permission at the top level as-needed?

Reply:
> Remediation steps

> Set top-level permissions as read-all or contents: read as described in GitHub's [documentation](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions).
Set any required write permissions at the job-level. Only set the permissions required for that job; do not set permissions: write-all at the job level.
To help determine the permissions needed for your workflows, you may use [StepSecurity's online tool](https://app.stepsecurity.io/secureworkflow/) by ticking the "Restrict permissions for GITHUB_TOKEN". You may also tick the "Pin actions to a full length commit SHA" to fix issues found by the Pinned-dependencies check

Reply:
So I think what I did here is sufficient